src/crate/testing/layer.py

```diff
@@ -277,11 +284,25 @@ def start(self):
             self._wait_for_master()
             sys.stderr.write('\nCrate instance ready.\n')
 
+    def out_thread_stopped(self):
+            return self._stop_out_thread.isSet()
+
+    def stop_out_thread(self):
+            self._stop_out_thread.set()
+
+    def consume(self, iterable):
+        while not self.out_thread_stopped():
```

Comment:
I think since the iterable is infinite/blocking (as it's linked to stdout of the subprocess) this condition will never be hit?